src/components/Dropdown.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React, { useState, useCallback } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from '../types/Person';
+
+type Props = {
+  peopleFromServer: Person[];
+  onSelected: (person: Person) => void;
+};
+
+export const Dropdown: React.FC<Props> = ({
+  peopleFromServer,
+  onSelected = () => {},
+}) => {
+  const [isActive, setIsActive] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(
+    debounce((value) => {
+      setAppliedQuery(value.trim() !== '' ? value : '');
+      setIsActive(true);
+    }, 1000),
+    [setAppliedQuery],
+  );
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newQuery = event.target.value;
+
+    setIsActive(false);
```

Comment:
`setIsActive` calls every time when you typing something. Could you optimize it?

Reply:
can you help me, I tried to use useRef, but it doesn't work?